src/command_modules/azure-cli-acs/azure/cli/command_modules/acs/custom.py

```diff
@@ -1583,10 +1590,12 @@ def aks_show(cmd, client, resource_group_name, name):
     return _remove_nulls([mc])[0]
 
 
-def aks_scale(cmd, client, resource_group_name, name, node_count, no_wait=False):
+def aks_scale(cmd, client, resource_group_name, name, node_count, nodepool_name="nodepool1", no_wait=False):
     instance = client.get(resource_group_name, name)
     # TODO: change this approach when we support multiple agent pools.
-    instance.agent_pool_profiles[0].count = int(node_count)  # pylint: disable=no-member
+    for agent_profile in instance.agent_pool_profiles:
+        if agent_profile.name == nodepool_name:
+            agent_profile.count = int(node_count)  # pylint: disable=no-member
```

Comment:
Cast error? Should the type of the parameter to be guaranteed in the parameter definition?

Reply:
I prefer not to change this right now, since i notice that this has been the case for different parameters, would it be ok if i add an issue and address it in a different PR?

Reply:
That will do.

Reply:
Thanks for the feedback, added a new issue https://github.com/Azure/azure-cli/issues/7388, will take care of it in next PR.